Code/Framework/AzToolsFramework/AzToolsFramework/ViewportSelection/EditorHelpers.cpp

```diff
@@ -173,6 +174,13 @@ namespace AzToolsFramework
             }
         }
 
+        // Allow entity system to override the selection
+        auto editorInteractionInterface = AZ::Interface<EditorInteractionInterface>::Get();
+        if (editorInteractionInterface != nullptr)
+        {
+            entityIdUnderCursor = editorInteractionInterface->RedirectEntitySelection(entityIdUnderCursor);
```

Comment:
After revising my prototype, I was thinking it may not be the best idea to allow any system to plug into the selection function. So I was moving towards creating an Editor system that handles the concept of container entity, and then have the appropriate checks be ran directly in the EditorHelpers flow.

Maybe we can discuss the approach in a bit?